pkg/specgen/generate/container.go

```diff
@@ -481,7 +489,68 @@ func ConfigToSpec(rt *libpod.Runtime, specg *specgen.SpecGenerator, contaierID s
 		}
 	}
 	specg.OverlayVolumes = overlay
-	specg.Mounts = conf.Spec.Mounts
+	name, mounts := c.SortUserVolumes(c.Spec())
+	tempInspect, err := c.GetMounts(name, conf.ImageVolumes, mounts)
+	if err != nil {
+		return nil, nil, err
+	}
+	mountsToPass := []string{}
+	for _, val := range tempInspect {
+		mountsToPass = append(mountsToPass, fmt.Sprintf("type=%v,src=%v,target=%v,%v", val.Type, val.Source, val.Destination, strings.Join(val.Options, ",")))
+	}
+	m, _, _, err := specgenutil.Mounts(mountsToPass)
+
+	// a lot of options get added later and are hard to translate.
+	// weed those out here
+	for err != nil && strings.Contains(err.Error(), "invalid mount option") {
+		errString := err.Error()
+		errString = errString[:strings.LastIndex(errString, ":")]
+		for i, val := range mountsToPass {
+			if strings.Contains(val, errString) {
+				ind := strings.Index(mountsToPass[i], errString) - 1
+				firstH := mountsToPass[i][0:ind]
+				secondH := mountsToPass[i][ind+len(errString)+1:]
+				mountsToPass[i] = firstH + secondH
+			}
+		}
```

Comment:
I have no idea what this is doing or why such weird manually conversion is needed. Please add at least a comment explaining what it does